### PR TITLE
Documentation for federated deployment

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @madhusudancs @nikhiljindal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1828)

### CODE REVIEW

**Comment by devin-donnelly** [comment]
Please retarget this PR to the release-1.5 branch for all 1.5 features.

File: docs/user-guide/federation/deployment.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
+by Kelsey Hightower, are also available to help you.
+
+You are also expected to have a basic
+[working knowledge of Kubernetes](/docs/getting-started-guides/) in
+general and [ReplicaSets](/docs/user-guide/replicasets/) in particular.
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: s/ReplicaSets/Deployments/

File: docs/user-guide/federation/deployment.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+## Overview
+
+Deployments in federation control plane (referred to as "federated Deployments" in
+this guide) are very similar to the traditional [Kubernetes
+ReplicaSets](/docs/user-guide/replicasets/), and provide the same functionality.
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: s/ReplicaSets/Deployments/

File: docs/user-guide/federation/deployment.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+Creating them in the federation control plane ensures that the desired number of
+replicas exist across the registered clusters.
+
+** Federated Deployment is currently in ALPHA, which means that the core functionality
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: If your intention is to make this bold, then I think you should remove the whitespace between ** Federated and yet. **

File: docs/user-guide/federation/deployment.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+``` shell
+kubectl --context=gce-asia-east1a get deployment mydep
+```
+
+The above assumes that you have a context named 'gce-asia-east1a'
+configured in your client for your cluster in that zone.
+
+These Deployments in underlying clusters will match the federation Deployment
+except in the number of replicas. Federation control plane will ensure that the
+sum of replicas in each cluster match the desired number of replicas in the
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: "... sum of replicas in all the clusters combined match the desired number of replicas ..."

File: docs/user-guide/federation/deployment.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+federation Deployment. 
+
+### Spreading Replicas in Underlying Clusters
+
+By default, replicas are spread equally in all the underlying clusters. For ex:
+if you have 3 registered clusters and you create a federated Deployment with
+`spec.replicas = 9`, then each Deployment in the 3 clusters will have
+`spec.replicas=3`.
+To modify the number of replicas in each cluster, you can specify
+[FederatedReplicaSetPreference](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/federation/apis/federation/types.go)
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: Just making sure, the annotation is still called `FederatedReplicaSetPreference` right? There is no `FederatedDeploymentPreference`?

File: docs/user-guide/federation/deployment.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+For example, you can do that using kubectl by running:
+
+```shell
+kubectl --context=federation-cluster delete deployment mydep
+```
+
+Note that at this point, deleting a federated Deployment will not delete the
+corresponding Deployments from underlying clusters.
+You must delete the underlying Deployments manually.
+We intend to fix this in the future.
```
Comment (madhusudancs) [inline docs/user-guide/federation/deployment.md]: Same comment as DaemonSets. We have cascading deletion for Federated DaemonSets now. I think you should point to - https://github.com/kubernetes/kubernetes.github.io/pull/1787/files#diff-74d1f4f6cca77a8c6f9631bbed28b69eR50 (whatever is the appropriate docs link for that).

File: docs/user-guide/federation/deployment.md (hunk 1 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: done.

File: docs/user-guide/federation/deployment.md (hunk 2 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: done.

File: docs/user-guide/federation/deployment.md (hunk 3 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

File: docs/user-guide/federation/deployment.md (hunk 5 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: No. We should rename the annotation and the structs.

File: docs/user-guide/federation/deployment.md (hunk 4 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

File: docs/user-guide/federation/deployment.md (hunk 6 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

**Comment by mwielgus** [comment]
Comments applied. PTAL.

**Comment by madhusudancs** [comment]
@mwielgus thanks! LGTM!

**Comment by madhusudancs** [comment]
cc @kubernetes/docs

File: docs/user-guide/federation/deployment.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+## Overview
+
+Deployments in federation control plane (referred to as "Federated Deployments" in
+this guide) are very similar to the traditional [Kubernetes
+Deployment](/docs/user-guide/deployment.md), and provide the same functionality.
+Creating them in the federation control plane ensures that the desired number of
+replicas exist across the registered clusters.
+
+**Federated Deployment is currently in ALPHA, which means that the core functionality
```
Comment (devin-donnelly) [inline docs/user-guide/federation/deployment.md]: "Currently" can get out of date. Reference the specific version instead.

"As of Kubernetes version 1.5, Federated Deployment is an Alpha feature. The core functionality of Deployment is present, but some features (such as full rollout compatibility) are still in development."

File: docs/user-guide/federation/deployment.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+---
+
+This guide explains how to use Deployments in the Federation control plane.
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes Cluster
```
Comment (devin-donnelly) [inline docs/user-guide/federation/deployment.md]: Please see the comments I made on the Prerequisites section here: https://github.com/kubernetes/kubernetes.github.io/pull/1827
and apply them as necessary to this document.

File: docs/user-guide/federation/deployment.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+``` shell
+kubectl --context=gce-asia-east1a get deployment mydep
+```
+
+The above assumes that you have a context named 'gce-asia-east1a'
+configured in your client for your cluster in that zone.
+
+These Deployments in underlying clusters will match the federation Deployment
+except in the number of replicas. Federation control plane will ensure that the
```
Comment (devin-donnelly) [inline docs/user-guide/federation/deployment.md]: Emphasize "except" with italics (*except*).

"Federation control plane will" -> "The Federation control plane ensures that the sum of replicas..."

**Comment by devin-donnelly** [comment]
Close; just needs a couple of fixes.

File: docs/user-guide/federation/deployment.md (hunk 8 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

File: docs/user-guide/federation/deployment.md (hunk 7 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

File: docs/user-guide/federation/deployment.md (hunk 9 above)
Comment (mwielgus) [inline docs/user-guide/federation/deployment.md]: Done.

**Pull request merged**